src/lib.rs

```diff
@@ -0,0 +1,79 @@
+use std::io;
+use rocket::serde::{Deserialize, Serialize};
+use rocket::serde::json::{Json};
+use rocket::http::{ContentType};
+use rocket::{Build, Rocket};
+use rocket::State;
+
+mod config;
+
+#[rocket::get("/")]
+fn index(root: &State<String>) -> Result<SuccessResponder<config::Config>, ErrorResponder> {
+    Ok(SuccessResponder::from(config::read_config(root)?))
+}
+
+pub fn api(root: String) -> Rocket<Build> {
+    rocket::build()
+        .manage(root)
+        .mount("/", rocket::routes![index])
+}
+
+#[derive(rocket::Responder)]
+#[response(status = 500, content_type = "json")]
```

Comment:
possibly we need a 400 responder here too:

```
$ curl -sv -H "Accept: application/json" localhost:8000/elsewhere  | jq
*   Trying ::1:8000...
* TCP_NODELAY set
* connect to ::1 port 8000 failed: Connection refused
*   Trying 127.0.0.1:8000...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8000 (#0)
> GET /elsewhere HTTP/1.1
> Host: localhost:8000
> User-Agent: curl/7.68.0
> Accept: application/json
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< content-type: application/json
< server: Rocket
< permissions-policy: interest-cohort=()
< x-content-type-options: nosniff
< x-frame-options: SAMEORIGIN
< content-length: 128
< date: Thu, 17 Nov 2022 09:48:22 GMT
< 
{ [128 bytes data]
* Connection #0 to host localhost left intact
{
  "error": {
    "code": 404,
    "reason": "Not Found",
    "description": "The requested resource could not be found."
  }
}
```

(without the header it returns some html, which is fine if a bit gross, but the format here is unexpected)

Reply:
ah yes, I see in the logs here:

```
GET /elsewhere application/json:
   >> No matching routes for GET /elsewhere application/json.
   >> No 404 catcher registered. Using Rocket default.
   >> Response succeeded.
```

Reply:
I figured that meant that on errors that we throw we'd return more sensible codes, but this I think belongs here - overriding the default 404 handler so that it matches the schema